### PR TITLE
mat64: use lapack64.Ormqr in Dense.QFromQR

### PR DESCRIPTION
**Pull request by vladimir-ch**

### CODE REVIEW

File: mat64/qr.go [hunk 1]
```diff
@@ -86,7 +86,7 @@ func (m *Dense) RFromQR(qr *QR) {
 
 // QFromQR extracts the m×m orthonormal matrix Q from a QR decomposition.
 func (m *Dense) QFromQR(qr *QR) {
-	r, c := qr.qr.Dims()
+	r, _ := qr.qr.Dims()
 	m.reuseAs(r, r)
```
Comment (kortschak) [inline mat64/qr.go]: I think the elements of m need to be zeroed for this use.

Comment (vladimir-ch) [inline mat64/qr.go]: It is, on lines 93-97.

Comment (kortschak) [inline mat64/qr.go]: Ah yes, hidden in the fold. As you'll have noticed, Exp uses an initial I as well and it's probably worth sharing the reuseAsZeroed between them.

Comment (kortschak) [inline mat64/qr.go]: I'll send that after this goes in.

**Comment by kortschak** [comment]
LGTM

**Comment by vladimir-ch** [comment]
I'll wait for @btracey .

**Comment by btracey** [comment]
Woo lapack!

LGTM

**Pull request merged**